### PR TITLE
Room Requests

### PR DESCRIPTION
**Pull request by HoeenCoder**
WIP

### CODE REVIEW

**Comment by ghost** [comment]
@HoeenCoder what is this for?

**Comment by HoeenCoder** [comment]
Room Policy rewrite, you cant buy rooms anymore.

**Comment by ghost** [comment]
So you just request one?

**Comment by HoeenCoder** [comment]
P much, though there will be limits, and restrictions, were working on them atm.

**Comment by ghost** [comment]
Oh cool

**Comment by ghost** [comment]
Now that I think about it, why isn't there a WIP Label???

File: spacialgaze-plugins/roomrequest.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		}
+		curRequest = {
+			name: Chat.escapeHTML(target[0].trim()),
+			type: toId(target[1]),
+			desc: Chat.escapeHTML(desc),
+			trusted: user.trusted,
+			staff: user.isStaff,
+			status: "pending",
+		};
+		Db.rooms.set(user.userid, curRequest);
```
Comment (DeathlyPlays) [inline spacialgaze-plugins/roomrequest.js]: Isn't that the db for autojoining rooms?

Comment (HoeenCoder) [reply spacialgaze-plugins/roomrequest.js]: no i made sure the db was empty on SG

Comment (DeathlyPlays) [reply spacialgaze-plugins/roomrequest.js]: Ah ok just checking

Comment (DeathlyPlays) [reply spacialgaze-plugins/roomrequest.js]: I always have to double check db files haha

**Comment by HoeenCoder** [comment]
Will finish travis later (1/3)*

File: spacialgaze-plugins/roomrequest.js (hunk 1 above)
Comment (Lord-Haji) [reply spacialgaze-plugins/roomrequest.js]: @DeathlyPlays SG never had the server side auto join feature

Comment (DeathlyPlays) [reply spacialgaze-plugins/roomrequest.js]: I was unaware @Lord-Haji

Comment (HoeenCoder) [reply spacialgaze-plugins/roomrequest.js]: We use the one i coded for main instead

Comment (DeathlyPlays) [reply spacialgaze-plugins/roomrequest.js]: Ah

**Pull request merged**